tests/test_postfix.py

```diff
@@ -0,0 +1,366 @@
+                    containerfile=CONTAINERFILE_POSTFIX_WITH_LDAP_ENABLED,
+                    extra_environment_variables={
+                        "SERVER_HOSTNAME": "localhost",
+                        "SMTP_RELAYHOST": "localhost",  # pointing to mailhog running locally
+                        "SMTP_PORT": "1025",
+                        "VIRTUAL_MBOX": "1",
+                        "VMAIL_UID": "5000",
+                        "USE_LDAP": "1",
+                        "LDAP_BASE_DN": "dc=example,dc=com",
+                        "LDAP_BIND_DN": "cn=admin,dc=example,dc=com",
+                        "LDAP_BIND_PASSWORD": "admin",
+                        "LDAP_USE_TLS": "0",
+                        "LDAP_TLS": "0",
+                        "LDAP_DOMAIN": "example.com",
+                        "LDAP_ADMIN_PASSWORD": "admin",
+                        "LDAP_CONFIG_PASSWORD": "config",
+                        "SETUP_FOR_MAILSERVER": "1",
+                        "MAIL_ACCOUNT_READER_PASSWORD": "admin",
+                        "LDAP_SERVER_URL": "ldap://localhost",
+                    },
+                    image_format=ImageFormat.DOCKER,
+                ),
+                marks=marks,
+            )
+        )
+
+    @pytest.mark.parametrize(
+        "container_per_test", POSTFIX_WITH_LDAP_ENABLED, indirect=True
+    )
+    def test_postfix_with_ldap_and_email_delivered(
+        container_per_test: ContainerData,
+    ) -> None:
+        # test relaying an email successfully with ldap lookup, using Postfix container
+
+        assert container_per_test.connection.run_expect(
+            [0], "nohup /usr/local/bin/mailhog > /dev/null 2>&1 &"
+        )
+
+        assert container_per_test.connection.run_expect(
+            [0], "/entrypoint/openldap-entrypoint.sh /usr/sbin/slapd"
+        )
+        assert "0.0.0.0:389" in container_per_test.connection.check_output(
+            "ss -tunlp"
+        )
+        assert container_per_test.connection.run_expect(
+            [0],
+            'ldapadd -x -D "cn=admin,dc=example,dc=com" -w "admin" -f /entrypoint/ldif/examples/example-user.ldif',
+        )
+
+        sendmail_cmd = 'echo "Subject: Test Email\n\nThis is a test email body." | sendmail -f user1@example.com user2@example.com'
+        assert container_per_test.connection.run_expect([0], sendmail_cmd)
+
+        assert (
+            "Mail queue is empty"
+            in container_per_test.connection.check_output("sleep 10; mailq")
+        )
+
+        for output in [
+            "from=<user1@example.com>",
+            "to=<user2@mail.example.com>, orig_to=<user2@example.com>",
+            "relay=localhost[127.0.0.1]:1025",
+            "dsn=2.0.0",
+            "status=sent (250 Ok: queued as",
+            "=@mailhog.example)",
+        ]:
+            assert output in container_per_test.connection.check_output(
+                "cat /var/log/postfix.log"
+            )
```

Comment:
I would suggest to reverse this, you're `cat`ing a (potentially large) file six times:
```suggestion
        log = container_per_test.connection.file("/var/log/postfix.log").content_string
        for output in [
            "from=<user1@example.com>",
            "to=<user2@mail.example.com>, orig_to=<user2@example.com>",
            "relay=localhost[127.0.0.1]:1025",
            "dsn=2.0.0",
            "status=sent (250 Ok: queued as",
            "=@mailhog.example)",
        ]:
            assert output in log
```

Alternatively, you can use [`file.contains`](https://testinfra.readthedocs.io/en/latest/modules.html#testinfra.modules.file.File.contains), but that relies on grep being installed. So I'd suggest to use the pure python solution.

Reply:
addressed.